src/rat/data_processing/metsim_input_processing.py

```diff
@@ -448,11 +448,17 @@ def _read_and_write_in_chunks(self):
                         log.debug(f"Found existing file at {self._outputpath} -- Updating in-place")
                         # Assuming the existing file structure is same as the one generated now. Basically
                         #   assuming that the previous file was also created by MetSimRunner
-                        existing = xr.open_dataset(self._outputpath)
-                        existing.close()
-                        last_existing_time = existing.time[-1]
-                        log.debug("Existing data: %s", last_existing_time)
-                        existing_to_append = existing.sel(time=slice(start_date_pd.to_datetime64() - np.timedelta64(120,'D') , last_existing_time))
+
```

Comment:
Same reason, for better readability and better coding practice. It just gives you better idea that why you need to load the dataset as you are using the data outside the for loop. Also, if you open NetCDF files using with statement, they will never give permission denied error (in case of Jupyter notebooks).